raft.go

```diff
@@ -1303,7 +1303,10 @@ func stepLeader(r *raft, m pb.Message) error {
 	case pb.MsgReadIndex:
 		// only one voting member (the leader) in the cluster
 		if r.prs.IsSingleton() {
-			if resp := r.responseToReadIndexReq(m, r.raftLog.committed); resp.To != None {
+			// It's possible that the "officially committed index" < "actually committed index",
```

Comment:
I think I understand what you mean, however the naming "officially commited index" vs "actually commited index" don't make sense for me. Commited index is just a commited index. I think should have a clearer explanation about difference between r.raftLog.commited vs r.prs.Commited()

Reply:
see the comment for method `readIndex`